test/source/tests/test_get.brs

```diff
@@ -134,3 +142,74 @@ End Function
 Function testCase_get_badPathDefault()
   return m.AssertEqual(m._.get({a:1}, invalid, "default"), "default")
 End Function
+
+Function testCase_get_simpleField()
+  source = CreateObject("roSGNode", "ContentNode")
+  source.id = "parentNode"
+  child1 = source.createChild("ContentNode")
+  child1.id = "abc"
+
+  value = m._.get(source, "id")
+  return m.AssertTrue(value = "parentNode")
+End Function
+
+Function testCase_get_simpleNode()
+  source = CreateObject("roSGNode", "ContentNode")
+  source.id = "parentNode"
+  child1 = source.createChild("ContentNode")
+  child1.id = "abc"
+
+  value = m._.get(source, "id")
+  return m.AssertTrue(value = "parentNode")
+End Function
+
+Function testCase_get_ChildNode()
+  source = CreateObject("roSGNode", "ContentNode")
+  source.id = "parentNode"
+  child1 = source.createChild("ContentNode")
+  child1.id = "abc"
+
+  value = m._.get(source, "0.id")
+  return m.AssertTrue(value = "abc")
+
+End Function
+
+Function testCase_get_defaultForNode()
+  source = CreateObject("roSGNode", "ContentNode")
+  source.id = "parentNode"
+  child1 = source.createChild("ContentNode")
+  child1.id = "abc"
+
+  value = m._.get(source, "0.0.id", "def")
+  return m.AssertEqual(value, "def")
+End Function
+
+Function testCase_get_badNode()
+  return m.AssertInvalid(m._.get(invalid, "id"))
+End Function
+
+Function testCase_get_badNodeDefault()
+  return m.AssertEqual(m._.get(invalid, "id", "default"), "default")
+End Function
+
+Function testCase_get_childInvalid()
+  source = CreateObject("roSGNode", "ContentNode")
+  source.id = "parentNode"
+
+  value = m._.get(source, "0.id")
+  return m.AssertInvalid(value)
+End Function
+
+Function testCase_get_deepChildInvalid()
+  source = CreateObject("roSGNode", "ContentNode")
+  source.id = "parentNode"
+  child1 = source.createChild("ContentNode")
+  child1.id = "abc"
+  child2 = source.createChild("ContentNode")
+  child2.id = "def"
+  grandchild1 = child2.createChild("ContentNode")
+  grandchild1.id = "ghi"
+
+  value = m._.get(source, "0.3.id")
+  return m.AssertInvalid(value)
+End function
```

Comment:
Can we please add tests for deep children that are valid. Can we also create tests for AAs and arrays on fields of source nodes and children nodes?